text/3349-cfg-os-version.md

```diff
@@ -0,0 +1,142 @@
+- Feature Name: `cfg-os-version`
+- Start Date: 2022-10-31
+- RFC PR: TBD
+- Rust Issue: TBD
+
+# Summary
+[summary]: #summary
+
+A new `cfg` key-value option `target_os_version`, and new predicates `os_version_eq`, `os_version_min`, and `os_version_range` that allow users to declare the primary (target-defined) API level required/supported by a block.  A second version of the predicates would take an additional key argument allowing targets to specify the versions of different OS components, e.g. kernel and libc versions.
```

Comment:
Can you add some more detail explaining the `target_os_version` key-value option somewhere? I don't see it explicitly described.

In particular, this RFC seems to introduce a novel syntax using dots. Could it maybe say why it is introducing that, instead of for example just underscores (like `target_os_version_windows`)? That could introduce some complications.

Reply:
Will do.  I have no strong opinions on the syntax for `target_os_version` so I'll present both options with some pros/cons listed for each.